test/groovy/CheckResultsPublishTest.groovy

```diff
@@ -0,0 +1,186 @@
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertTrue
+
+import util.JenkinsConfigRule
+import util.JenkinsSetupRule
+
+class CheckResultsPublishTest extends BasePipelineTest {
+    Map publisherStepOptions
+    List archiveStepPatterns
+
+    @Rule
+    public RuleChain ruleChain = RuleChain.outerRule(new JenkinsSetupRule(this)).around(new JenkinsConfigRule(this))
+
+    def stepUnderTest
```

Comment:
in order to keep it in synch with the other tests I would suggest to use rather `checkResultsPublishScript`